biofefi/machine_learning/nn_models.py

```diff
@@ -0,0 +1,271 @@
+import numpy as np
+import pandas as pd
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from sklearn.base import BaseEstimator, ClassifierMixin, RegressorMixin
+
+from biofefi.machine_learning.nn_networks import BaseNetwork
+from biofefi.options.enums import ModelNames, OptimiserTypes, ProblemTypes
+from biofefi.options.ml import BrnnOptions
+
+
+class BayesianRegularisedNNClassifier(BaseNetwork, BaseEstimator, ClassifierMixin):
+    """
+    This class defines a Bayesian Regularized Neural
+    Network for classification tasks.
+
+    Args:
+        problem_type (ProblemTypes): The type of problem
+        (Classification).
+        - layer1: The first linear layer of the network.
+        - layer2: The second linear layer of the network.
+        - output_layer: The output linear layer of the network.
+    """
+
+    def __init__(self, problem_type: ProblemTypes):
+        """
+        Initializes the BayesianRegularisedNNClassifier class.
+        """
+        super().__init__()
+        self._name = ModelNames.BRNNClassifier
+        self.problem_type = problem_type
+        self.layer1, self.layer2, self.output_layer = None, None, None
+
+    def _initialize_network(self, input_dim, output_dim):
+        """
+        Initializes the network layers based on the input
+        and output dimensions.
+
+        Args:
+            input_dim (int): The input dimension of the data.
+            output_dim (int): The output dimension of the
+            data, determined dynamically.
+        """
+        # Define hidden layers and output layer
+        self.layer1 = nn.Linear(input_dim, BrnnOptions.hidden_dim)
+        self.layer2 = nn.Linear(BrnnOptions.hidden_dim, BrnnOptions.hidden_dim)
+        self.output_layer = nn.Linear(BrnnOptions.hidden_dim, output_dim)
```

Comment:
Add an instance of BrnnOptions to the constructor and use that here. BrnnOptions is a class

Reply:
Ok